CI is failing on `main` due to flake8

See https://github.com/posit-dev/py-shiny/actions/runs/7538643088/job/20519572273 for example

```
flake8 --show-source .
./shiny/_utils.py:396:13: B036 Don't except `BaseException` unless you plan to re-raise it.
            except BaseException as e:
            ^
./shiny/_utils.py:422:9: B036 Don't except `BaseException` unless you plan to re-raise it.
        except BaseException as e:
        ^
./shiny/render/renderer/_renderer.py:228:13: B037 Class `__init__` methods must not return or yield and any values.
            return cast(AsyncValueFn[IT], fn)
            ^
```

--- a/shiny/_utils.py
+++ b/shiny/_utils.py
@@ -393,7 +393,7 @@ def run_coro_hybrid(coro: Awaitable[R]) -> "asyncio.Future[R]":
             assert fut.done()
             try:
                 fut.result()
-            except BaseException as e:
+            except BaseException as e:  # noqa: B036
                 exc = e
 
         if result_future.cancelled():
@@ -419,7 +419,7 @@ def run_coro_hybrid(coro: Awaitable[R]) -> "asyncio.Future[R]":
         except (KeyboardInterrupt, SystemExit) as e:
             result_future.set_exception(e)
             raise
-        except BaseException as e:
+        except BaseException as e:  # noqa: B036
             result_future.set_exception(e)
         else:
             # If we get here, the coro didn't finish. Schedule it for completion.

--- a/shiny/render/renderer/_renderer.py
+++ b/shiny/render/renderer/_renderer.py
@@ -225,7 +225,9 @@ class AsyncValueFn(Generic[IT]):
 
     def __init__(self, fn: Callable[[], IT] | Callable[[], Awaitable[IT]]):
         if isinstance(fn, AsyncValueFn):
-            return
+            raise TypeError(
+                "Must not call `AsyncValueFn.__init__` with an object of class `AsyncValueFn`"
+            )
         self._is_async = is_async_callable(fn)
         self._fn = wrap_async(fn)
         self._orig_fn = fn

--- a/shiny/render/renderer/_renderer.py
+++ b/shiny/render/renderer/_renderer.py
@@ -225,7 +225,7 @@ class AsyncValueFn(Generic[IT]):
 
     def __init__(self, fn: Callable[[], IT] | Callable[[], Awaitable[IT]]):
         if isinstance(fn, AsyncValueFn):
-            return cast(AsyncValueFn[IT], fn)
+            return
         self._is_async = is_async_callable(fn)
         self._fn = wrap_async(fn)
         self._orig_fn = fn

